src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -1411,6 +1390,56 @@ public static void Main()
             CompileAndVerify(compilation, expectedOutput: "-10121");
         }
 
+        [Theory]
+        [InlineData(
+            "{ null, null, new(0, 0) }",
+            "[..{ Length: >=2 }, { X: 0, Y: 0 }]",
+            "e.Length, e[0..^1], e[0..^1].Length, e[^1], e[^1].X, e[^1].Y, True")]
+        [InlineData(
+            "{ null, null, new(0, 0) }",
+            "[.., { X: 0, Y: 0 }]",
+            "e.Length, e[^1], e[^1].X, e[^1].Y, True")]
+        [InlineData(
+            "{ new(0, 5) }",
+            "[.., { X:0, Y:0 }] or [{ X:0, Y:5 }]",
+            "e.Length, e[^1], e[^1].X, e[^1].Y, e[0], e[0].X, e[0].Y, True")]
```

Comment:
Given this result, I think I'd like to see `{ null, new(0, 5) }` tested with an `or` pattern after a slice.

Reply:
I'm not clear where "after a slice" points to. Can you give an example?

Reply:
This same pattern, but matching on a different input.

Reply:
To be clear:
```
        [InlineData(
            "{ new(0, 1), new(0, 5) }",
            "[.., { X:0, Y:0 }] or [{ X:0, Y:5 }]",
            "e.Length, e[^1], e[^1].X, e[^1].Y, e[0], e[0].X, e[0].Y, True")]
```